src/Elastic.Markdown/Myst/InlineParsers/DiagnosticLinkInlineParser.cs

```diff
@@ -108,9 +108,20 @@ public override bool Match(InlineProcessor processor, ref StringSlice slice)
 
 		if (link.FirstChild == null || !string.IsNullOrEmpty(anchor))
 		{
-			var file = string.IsNullOrWhiteSpace(url) ? context.Path
-				: context.Build.ReadFileSystem.FileInfo.New(Path.Combine(context.Build.SourcePath.FullName, url.TrimStart('/')));
+			var file = string.IsNullOrWhiteSpace(url)
+				? context.Path
+				: url.StartsWith('/')
+					? context.Build.ReadFileSystem.FileInfo.New(Path.Combine(context.Build.SourcePath.FullName, url.TrimStart('/')))
+					: context.Build.ReadFileSystem.FileInfo.New(Path.Combine(context.Path.Directory!.FullName, url));
 			var markdown = context.GetDocumentationFile?.Invoke(file) as MarkdownFile;
+			if (markdown == null)
+			{
+				processor.EmitWarning(line,
+					column,
+					length,
+					$"Can not create an auto title link to '{file.FullName}' however this file can not be resolved.");
```

Comment:
I don't fully understand the warning. It sounds a bit contradicting.

> can not create ... however .. can not resolve ?

else LGTM

Reply:
You are right this is a terrible error message amending the PR now :)